src/transform/visitor.ts

```diff
@@ -0,0 +1,1259 @@
+import { createObjectTypeProperty } from '../ast/types/object-type-property';
+import { createOpaqueType } from '../ast/types/opaque-type';
+import { createQualifiedType } from '../ast/types/qualified-type';
+import { createTypeInstantiations } from '../ast/types/type-instantiations';
+import { createTypeAnnotation } from '../ast/types/type-annotation';
+import { createTypeParameterDeclaration } from '../ast/types/type-parameter-declaration';
+import { createTypeParameterInstantiation } from '../ast/types/type-parameter-instantiation';
+import { createTypeParameterList } from '../ast/types/type-parameter-list';
+import { createTypeofType } from '../ast/types/typeof-type';
+import { createTypeReference } from '../ast/types/type-reference';
+import { createFunctionType } from '../ast/types/function-type';
+import { createIndexedAccessType } from '../ast/types/indexed-access-type';
+import { createIntersectionType } from '../ast/types/intersection-type';
+import { createOptionalType } from '../ast/types/optional-type';
+import { createParenthesizedType } from '../ast/types/parenthesized-type';
+import { createRestType } from '../ast/types/rest-type';
+import { createTupleType } from '../ast/types/tuple-type';
+import { createTypeParameter } from '../ast/types/type-parameter';
+import { createUnionType } from '../ast/types/union-type';
+import { createDecoratorList } from '../ast/expressions/decorator-list';
+import { createDecorator } from '../ast/expressions/decorators';
+import { createFormalParameterList } from '../ast/expressions/formal-parameter-list';
+import { createImportClause } from '../ast/module/import-clause';
+import { createNamespaceExportDeclaration } from '../ast/module/namespace-export-declaration';
+import { createNameSpaceImport } from '../ast/module/namespace-import';
+import { createExportDefault } from '../ast/module/export-default';
+import { createExportDeclaration } from '../ast/module/export-declaration';
+import { createExportFromClause } from '../ast/module/export-from-clause';
+import { createExportSpecifier } from '../ast/module/export-specifier';
+import { createExportsList } from '../ast/module/exports-list';
+import { createImportDeclaration } from '../ast/module/import-declaration';
+import { createImportSpecifier } from '../ast/module/import-specifier';
+import { createImportsList } from '../ast/module/imports-list';
+import { createNamedExports } from '../ast/module/named-exports';
+import { createNamedImports } from '../ast/module/named-imports';
+import { createSpreadElement } from '../ast/expressions/spread-element';
+import {
+  Transform,
+  LexicalEnvironmentFlags,
+  createNodeArray,
+  extractSingleNode,
+  startLexicalEnvironment,
+  endLexicalEnvironment,
+  concatenate
+} from './core';
+
+export function visitEachChild(transform: Transform, node: any, visitor: (node: SyntaxKind) => SyntaxKind): any {
+  const kind = node.kind;
```

Comment:
have you tested it? the type node: SyntaxKind seems not correct.

Reply:
I'm working on testing and improve it locale. There are bugs that I fixed already, but not pushed

Reply:
The SyntaxKind is correct, you need to do a type cast to get it correct. Se my update